lib/puppet/network/http/webrick/rest.rb

```diff
@@ -15,8 +15,12 @@ def self.mutex
 
   def initialize(server)
     raise ArgumentError, "server is required" unless server
-    register([Puppet::Network::HTTP::API::V3.routes,
-              Puppet::Network::HTTP::API::V2.routes])
+    prefix = Regexp.new("^#{Puppet[:master_url_prefix]}")
```

Comment:
We chatted at some point about how we might need to include a trailing slash in the regex in order to make sure it falls through to the 404 properly.  Is that something that only affects the subsequent PR?  Or did you find a different solution?